src/main/java/net/dries007/tfc/common/blocks/wood/TFCChestBlock.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed under the EUPL, Version 1.2.
+ * You may obtain a copy of the Licence at:
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ */
+
+package net.dries007.tfc.common.blocks.wood;
+
+import java.util.function.Supplier;
+import javax.annotation.Nullable;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.ChestBlock;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraft.world.level.block.entity.BlockEntityTicker;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+import net.minecraft.world.level.block.entity.ChestBlockEntity;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraftforge.registries.RegistryObject;
+
+import net.dries007.tfc.common.blockentities.TFCBlockEntities;
+import net.dries007.tfc.common.blockentities.TFCChestBlockEntity;
+import net.dries007.tfc.common.blocks.EntityBlockExtension;
+import net.dries007.tfc.common.blocks.ExtendedProperties;
+import net.dries007.tfc.common.blocks.IForgeBlockExtension;
+
+public class TFCChestBlock extends ChestBlock implements IForgeBlockExtension, EntityBlockExtension
+{
+    private final String textureLocation;
+    private final ExtendedProperties extendedProperties;
+
+
+    public TFCChestBlock(ExtendedProperties properties, String textureLocation)
+    {
+        this(properties, textureLocation, TFCBlockEntities.CHEST::get); // () -> x.get() passes type check but casting (Sup<BEType<ChestBE>>) fails. Thanks compiler
```

Comment:
Can you not just pass `TFCBlockEntities.CHEST` ? The `RegistryObject<T>` itself implements `Supplier<T>` and the generic type bound `BlockEntityType<? extends ChestBlockEntity>` should be satisfied.